extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -182,6 +189,10 @@ public void build(
 
         Collection<AnnotationInstance> applicationPaths = index.getAnnotations(ResteasyDotNames.APPLICATION_PATH);
 
+        if (resteasyConfig.ignoreApplicationClasses) {
+            applicationPaths = Collections.emptySet();
+        }
```

Comment:
The `index.getAnnotations` call above can be avoided if `resteasyConfig.ignoreApplicationClasses` is `true`